### PR TITLE
Add type parameter to publish resources api, add the ability to overlay/underlay a fetched image 

### PR DESCRIPTION
**Pull request by RTLcoil**

### CODE REVIEW

File: Cloudinary.Test/ApiTest.cs [hunk 1]
```diff
@@ -416,6 +418,24 @@ public void testOverlayError2()
...
+            //http://image.com/img/seatrade_supplier_logo.jpg
+
+            //fetch image
+            //http://image.com/files/8813/5551/7470/cruise-ship.png
+
+            var transformation = new Transformation().Overlay(new FetchLayer().Url("http://image.com/img/seatrade_supplier_logo.jpg"));
+            var uri = m_api.UrlImgFetch.Transform(transformation).BuildUrl("http://image.com/files/8813/5551/7470/cruise-ship.png");
+            Assert.AreEqual(m_defaultImgFetchPath + "l_fetch:aHR0cDovL2ltYWdlLmNvbS9pbWcvc2VhdHJhZGVfc3VwcGxpZXJfbG9nby5qcGc=/http://image.com/files/8813/5551/7470/cruise-ship.png", uri);
+
+            //var transformation = new Transformation().Overlay(new FetchLayer().PublicId("test").Url("http:\\sampleuri")).Width(200);
```
Comment (ronen-burin) [inline Cloudinary.Test/ApiTest.cs]: Why do we have these comments? if not needed - let's remove..

File: Cloudinary/Shared/Coudinary.NetCoreShared/Transforms/FetchLayer.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace CloudinaryDotNet
+{
+    public class FetchLayer: BaseLayer<FetchLayer>
+    {
+        protected string m_url;
+
+        public FetchLayer()
+        {
+            m_resourceType = "fetch";
```
Comment (ronen-burin) [inline Cloudinary/Shared/Coudinary.NetCoreShared/Transforms/FetchLayer.cs]: Maybe use const?

File: Cloudinary/Shared/Coudinary.NetCoreShared/ApiShared.cs [hunk 3]
```diff
@@ -165,6 +165,21 @@ public Url UrlImgUp
...
         /// <summary>
+        /// Default URL for working with fetched images
+        /// </summary>
+        public Url UrlImgFetch
+        {
+            get
+            {
+                return Url
+                    .ResourceType("image")
+                    .Action("fetch")
```
Comment (ronen-burin) [inline Cloudinary/Shared/Coudinary.NetCoreShared/ApiShared.cs]: Maybe use const?

File: Cloudinary.Test/ApiTest.cs (hunk 1 above)
Comment (RTLcoil) [inline Cloudinary.Test/ApiTest.cs]: You're right, we'll remove it.

File: Cloudinary/Shared/Coudinary.NetCoreShared/Transforms/FetchLayer.cs (hunk 2 above)
Comment (RTLcoil) [inline Cloudinary/Shared/Coudinary.NetCoreShared/Transforms/FetchLayer.cs]: Yup, we'll use const.

File: Cloudinary/Shared/Coudinary.NetCoreShared/ApiShared.cs (hunk 3 above)
Comment (RTLcoil) [inline Cloudinary/Shared/Coudinary.NetCoreShared/ApiShared.cs]: Sure.

File: Cloudinary.Test/ApiTest.cs (hunk 1 above)
Comment (RTLcoil) [inline Cloudinary.Test/ApiTest.cs]: Included in PR 51

File: Cloudinary/Shared/Coudinary.NetCoreShared/Transforms/FetchLayer.cs (hunk 2 above)
Comment (RTLcoil) [inline Cloudinary/Shared/Coudinary.NetCoreShared/Transforms/FetchLayer.cs]: Included in PR 51

File: Cloudinary/Shared/Coudinary.NetCoreShared/ApiShared.cs (hunk 3 above)
Comment (RTLcoil) [inline Cloudinary/Shared/Coudinary.NetCoreShared/ApiShared.cs]: Included in PR 51

**Pull request closed without merging**